Helper/Customer.php

```diff
@@ -115,29 +115,42 @@ public function getCustomVariables($object, $data = [])
 
     /**
      * To get Customer model.
-     * 
-     * @param  string $param
-     * 
+     *
+     * @param string $param
+     *
      * @return \Magento\Customer\Model\Customer
+     *
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
      */
     public function getObject($param)
     {
         $this->customerModel->setWebsiteId($this->storeManager->getStore()->getWebsiteId());
-        $this->customerModel->loadByEmail($param);
+        try {
+            $this->customerModel->loadByEmail($param);
+        } catch (\Exception $e) {
+            throw new \Exception($e->getTraceAsString());
+        }
```

Comment:
What is the value add here by re-throwing an exception? Aren't you losing some of the exception info by just getting the trace?

Reply:
@giraclion Not sure my comment here was clear the first time. In general you should not be catching an exception unless you intend to do something meaningful. There are a few exceptions and this post describes one opinion on what they are: https://stackoverflow.com/a/5551828. The code above looks like an anti-pattern to me. I don't see the reason for catching the exception.

Reply:
@lindsaycooper I can remove it, not a problem. Issue was that in case of any exception there, Magento doesn't not record a trace, only highest level exception text, which in most cases doesn't help. It was added more for debug reasons, in any case fronted is not exposed in this particular case

Reply:
That makes sense to leave in then.

Reply:
@giraclion That makes sense to leave it as you have it then. Thanks for clarifying.